### PR TITLE
feat: make getElementTree composable

### PR DESCRIPTION
**Pull request by neatbyte-vnobis**
## Changes
Make getElementTree  composable. Filter out ref blocks from Navigator tree.

## How Has This Been Tested?
Manual

## Documentation
None

### CODE REVIEW

File: packages/app-page-builder/src/types.ts
Comment (Pavel910) [inline packages/app-page-builder/src/types.ts]: @neatbyte-vnobis is there a reason to convert this to a `props` object? I must be missing something, but it doesn't look like you're using anything other than the `element`, so not sure you need this 🤔 

And secondary, why `Promise<any>`? If you have a different form of `element` coming back from `getElementTree`, maybe it's better to add support for generic type on that function.

**Review by Pavel910 — COMMENTED** [review]
@neatbyte-vnobis I left a comment, want to clarify something before approval or requesting changes. Thanks!

File: packages/app-page-builder/src/types.ts
Comment (neatbyte-vnobis) [reply packages/app-page-builder/src/types.ts]: @Pavel910 I have updated the types

Not only `element` is used (in recursion also `path` is passed).
Due to composeAsync inner working only one parameter can be passed, so `getElementTree(element, path)` was replaced with `getElementTree({ element, path })`.

I have created a new type inside types (as there were multiple definitions with different interfaces in types and EventActionHandlerProvider.tsx)

And yes, no need for `any` there

**Review by Pavel910 — APPROVED** [review]
@neatbyte-vnobis thanks for clarification, makes sense.

**Pull request merged**
